Automattic/jetpack
# Add Social Previews to Jetpack Social

#### Changes proposed in this Pull Request:

This PR adds Social Previews to Jetpack Social. The existing components in the Jetpack plugin are moved to the `publicize-components` package, and all references to upgrading are removed. In a follow-up PR, we'll use these components in the Jetpack plugin as well, without the upgrade nudges.

#### Other information:


#### Jetpack product discussion


1202807135507995-as-1202862455837859/f

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






* Make sure your testing site has Jetpack Social active (not the Jetpack plugin) and is connected to Jetpack.
* Start writing a new post.
* Confirm that the previews section is visible in the sidebar.
* Confirm that the previews modal and the different social media platforms display correctly.
* Press Publish, and confirm that the previews are visible in the pre-publish section as well and work correctly.

<img width="815" alt="image" src="https://user-images.githubusercontent.com/1713699/187461986-ffa30c87-7afd-4c6e-90a0-e04b3596ba18.png">
<img width="822" alt="image" src="https://user-images.githubusercontent.com/1713699/187461996-46fdfa36-e002-4cf9-90d5-1a15d0485e6f.png">
<img width="957" alt="image" src="https://user-images.githubusercontent.com/1713699/187462010-d3230e8f-1b62-4d4c-940a-ff552e5dc8b8.png">
